### PR TITLE
Create 6-InterfaceListItem.yaml and reference interfaces in 2-ComponentListItem.yaml

### LINKED ISSUE #1: extend ComponentListItem definition to include interfaces
where interfaces are directly owned by a part or tool

## References
- http://www.opencaesar.io/oml-tutorials/#tutorial2-p6

### PR DESCRIPTION
**Pull request by kchiu002**
Close Issue #1

### CODE REVIEW

File: architecture/3-DataStructures/6-InterfaceListItem.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Component List Item
```
Comment (capsulecorplab) [inline architecture/3-DataStructures/6-InterfaceListItem.yaml]: `Component List Item` should say `Interface List Item`

File: architecture/3-DataStructures/6-InterfaceListItem.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+purpose: Identifies an interface on a part or tool.
+template: |
+  id: {{id}}
+  description: {{description}}
+  descriptionLong: {{descriptionLong}}
+elements:
+  - key: id
+    type: string
+    itemType: ''
+    description: Part's ID (or part number).  Format = single word, only lowercase letters, and may contain hyphens and underscores.
```
Comment (capsulecorplab) [inline architecture/3-DataStructures/6-InterfaceListItem.yaml]: `Part's ID (or part number)` should be `Interface ID`

File: architecture/3-DataStructures/6-InterfaceListItem.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    derivedFrom: []
+  - key: description
+    type: string
+    itemType: ''
+    description: Human readable name for the part (not name of the selected component for this part).
```
Comment (capsulecorplab) [inline architecture/3-DataStructures/6-InterfaceListItem.yaml]: `part` should be `interface`

File: architecture/3-DataStructures/2-ComponentListItem.yaml [hunk 4]
```diff
@@ -56,4 +57,9 @@ elements:
     itemType: ''
     description: Developer comments on this part or tool.
     derivedFrom: []
+  - key: interfaces
+    type: list
+    itemType: 3-DataStructures/6-InterfaceListItem.yaml'
+    description: Listing the interfaces of this part or tool.
```
Comment (capsulecorplab) [inline architecture/3-DataStructures/2-ComponentListItem.yaml]: I would reword `Listing the interfaces of this part or tool.` to `Listing of interfaces for the part or tool.`

**Review by capsulecorplab — CHANGES_REQUESTED** [review]
some minor changes

File: architecture/3-DataStructures/6-InterfaceListItem.yaml (hunk 3 above)
Comment (capsulecorplab) [reply architecture/3-DataStructures/6-InterfaceListItem.yaml]: note: there are two instances of the word `part` on this line

File: architecture/3-DataStructures/6-InterfaceListItem.yaml (hunk 1 above)
Comment (kchiu002) [reply architecture/3-DataStructures/6-InterfaceListItem.yaml]: Resolved [4cbcc20](https://github.com/ODU-CGA-CubeSat/dof-cubesat/pull/2/commits/4cbcc2047b672037edb01d55d017f9938b041063)

File: architecture/3-DataStructures/6-InterfaceListItem.yaml (hunk 2 above)
Comment (kchiu002) [reply architecture/3-DataStructures/6-InterfaceListItem.yaml]: Resolved [4cbcc20](https://github.com/ODU-CGA-CubeSat/dof-cubesat/pull/2/commits/4cbcc2047b672037edb01d55d017f9938b041063)

File: architecture/3-DataStructures/6-InterfaceListItem.yaml (hunk 3 above)
Comment (kchiu002) [reply architecture/3-DataStructures/6-InterfaceListItem.yaml]: Resolved [4cbcc20](https://github.com/ODU-CGA-CubeSat/dof-cubesat/pull/2/commits/4cbcc2047b672037edb01d55d017f9938b041063)

File: architecture/3-DataStructures/2-ComponentListItem.yaml (hunk 4 above)
Comment (kchiu002) [reply architecture/3-DataStructures/2-ComponentListItem.yaml]: Resolved [4cbcc20](https://github.com/ODU-CGA-CubeSat/dof-cubesat/pull/2/commits/4cbcc2047b672037edb01d55d017f9938b041063)

**Review by capsulecorplab — APPROVED** [review]
LGTM
